nautobot/core/utils/lookup.py

```diff
@@ -218,3 +220,125 @@ def get_created_and_last_updated_usernames_for_model(instance):
+        base_path (str): String to prepend to all URL patterns yielded.
+            Default if unspecified is the string `"/"`.
+
+    Yields:
+        (str): Each URL pattern defined in the given urlconf and its descendants
+
+    Examples:
+        >>> generator = get_url_patterns()
+        >>> next(generator)
+        '/'
+        >>> next(generator)
+        '/search/'
+        >>> next(generator)
+        '/login/'
+        >>> next(generator)
+        '/logout/'
+        >>> next(generator)
+        '/circuits/circuits/<uuid:pk>/terminations/swap/'
+
+        >>> import example_plugin.urls as example_urls
+        >>> for url_pattern in get_url_patterns(example_urls, base_path="/plugins/example-plugin/"):
+        ...     print(url_pattern)
+        ...
+        /plugins/example-plugin/
+        /plugins/example-plugin/config/
+        /plugins/example-plugin/models/<uuid:pk>/dynamic-groups/
+        /plugins/example-plugin/other-models/<uuid:pk>/dynamic-groups/
+        /plugins/example-plugin/docs/
+        /plugins/example-plugin/circuits/<uuid:pk>/example-plugin-tab/
+        /plugins/example-plugin/devices/<uuid:pk>/example-plugin-tab-1/
+        /plugins/example-plugin/devices/<uuid:pk>/example-plugin-tab-2/
+        /plugins/example-plugin/override-target/
+        /plugins/example-plugin/^models/$
+        /plugins/example-plugin/^models/add/$
+        /plugins/example-plugin/^models/import/$
+        /plugins/example-plugin/^models/edit/$
+        /plugins/example-plugin/^models/delete/$
+        /plugins/example-plugin/^models/all-names/$
+        /plugins/example-plugin/^models/(?P<pk>[^/.]+)/$
+        /plugins/example-plugin/^models/(?P<pk>[^/.]+)/delete/$
+        /plugins/example-plugin/^models/(?P<pk>[^/.]+)/edit/$
+        /plugins/example-plugin/^models/(?P<pk>[^/.]+)/changelog/$
+        /plugins/example-plugin/^models/(?P<pk>[^/.]+)/notes/$
+        /plugins/example-plugin/^other-models/$
+        /plugins/example-plugin/^other-models/add/$
+        /plugins/example-plugin/^other-models/edit/$
+        /plugins/example-plugin/^other-models/delete/$
+        /plugins/example-plugin/^other-models/(?P<pk>[^/.]+)/$
+        /plugins/example-plugin/^other-models/(?P<pk>[^/.]+)/delete/$
+        /plugins/example-plugin/^other-models/(?P<pk>[^/.]+)/edit/$
+        /plugins/example-plugin/^other-models/(?P<pk>[^/.]+)/changelog/$
+        /plugins/example-plugin/^other-models/(?P<pk>[^/.]+)/notes/$
+    """
+    if urlconf is None:
+        urlconf = settings.ROOT_URLCONF
+    if patterns_list is None:
+        patterns_list = get_resolver(urlconf).url_patterns
+
+    for item in patterns_list:
+        if isinstance(item, URLPattern):
+            yield base_path + str(item.pattern)
+        elif isinstance(item, URLResolver):
+            # Recurse!
+            yield from get_url_patterns(urlconf, item.url_patterns, base_path + str(item.pattern))
+
+
+def get_url_for_url_pattern(url_pattern):
```

Comment:
Should this be `get_valid_url_for_url_pattern` or `get_fake_url_for_url_pattern` maybe?

Reply:
I'd be fine with either.